xonsh/events.py

```diff
@@ -178,8 +187,22 @@ def discard(self, item):
         self._fired.discard(item)
         self._unfired.discard(item)
 
+    def _call(self, handler):
+        try:
+            handler(*self._pargs, **self._kwargs)
+        except Exception:
+            print_exception("Exception raised in event handler; ignored.")
+
     def fire(self, *pargs, **kwargs):
-        raise NotImplementedError("See #1550")
+        if self._hasfired:
+            return
```

Comment:
Well, if each handler is only callable once, how do you handle multiple firings? What does that even mean?

A reset method is easy (merge fired into unfired, reset fired, clear hasfired), but I'm going to claim YAGNI in this case. I can't imagine a time you'd need it except _maybe_ for testing.